cli/src/pcluster/templates/traditional_scheduler_builder.py

```diff
@@ -0,0 +1,98 @@
+# Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+# with the License. A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+# OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions and
+# limitations under the License.
+
+from aws_cdk import aws_dynamodb as dynamodb
+from aws_cdk import aws_iam as iam
+from aws_cdk.core import CfnDeletionPolicy, Construct, Stack
+
+from pcluster.constants import PCLUSTER_DYNAMODB_PREFIX
+from pcluster.templates.cdk_builder_utils import create_hash_suffix
+
+
+class TraditionalSchedulerConstruct(Construct):
+    """Create the resources required when using a tradition scheduler (e.g. slurm or plugin, but not awsbatch)."""
+
+    def __init__(
+        self,
+        scope: Construct,
+        id: str,
+        stack_name: str,
+        managed_head_node_instance_role: iam.CfnRole,
+    ):
+        super().__init__(scope, id)
+        self.stack_scope = scope
+        self.stack_name = stack_name
+        self.managed_head_node_instance_role = managed_head_node_instance_role
+
+        self._add_resources()
+
+    # -- Utility methods --------------------------------------------------------------------------------------------- #
+
+    def _format_arn(self, **kwargs):
+        return Stack.of(self).format_arn(**kwargs)
+
+    # -- Parameters -------------------------------------------------------------------------------------------------- #
+
+    # -- Resources --------------------------------------------------------------------------------------------------- #
+
+    def _add_resources(self):
+        # DynamoDB to store compute fleet status
+        self._add_dynamodb_table()
+
+        # Add Traditional Scheduler Policies to new instances roles
+        if self.managed_head_node_instance_role:
+            self._add_policies_to_head_node_role("HeadNode", self.managed_head_node_instance_role.ref)
```

Comment:
we could manage the policy in `HeadNodeIamResources` as already done for the other policies and eventually also get rid of this class if only used for dynamo db table creation.

Reply:
moved into HeadNodeIamResources